localflavor/cu/forms.py

```diff
@@ -0,0 +1,188 @@
+    A Select widget that uses a list of cuban regions as its choices.
+
+    .. versionadded:: 1.6
+    """
+
+    def __init__(self, attrs=None):
+        super(CURegionSelect, self).__init__(attrs, choices=REGION_CHOICES)
+
+
+class CUProvinceField(Field):
+    """
+    A form field for a cuban province.
+    The input is validated against a dictionary which includes names and abbreviations.
+
+    It normalizes the input to the standard abbreviation for the given province.
+
+    .. versionadded:: 1.6
+    """
+
+    default_error_messages = {
+        'invalid': _('Enter a cuban province.'),
+    }
+
+    def clean(self, value):
+        super(CUProvinceField, self).clean(value)
+        if value in EMPTY_VALUES:
+            return ''
+        try:
+            value = value.strip().lower()
+        except AttributeError:
+            pass
+        else:
+            try:
+                return PROVINCE_NORMALIZED[value.strip().lower()]
+            except KeyError:
+                pass
+        raise ValidationError(self.error_messages['invalid'])
+
+
+class CUProvinceSelect(Select):
+    """
+    A Select widget that uses a list of cuban provinces as its choices.
+
+    .. versionadded:: 1.6
+    """
+
+    def __init__(self, attrs=None):
+        super(CUProvinceSelect, self).__init__(attrs, choices=PROVINCE_CHOICES)
+
+
+class CUPostalCodeField(RegexField):
+    """
+    A form field for a cuban Zip Code.
+
+    Taken from : http://mapanet.eu/Postal_Codes/?C=CU
+
+    The cuban ZIP code is a combination of 5 digits non begin with 0.
+
+    .. versionadded:: 1.6
+    """
+
+    default_error_messages = {
+        'invalid': _('Enter a valid zip code in the format XXXXX.'),
+    }
+
+    def __init__(self, *args, **kwargs):
+        super(CUPostalCodeField, self).__init__(r'^[1-9]\d{4}$', *args, **kwargs)
+
+    def to_python(self, value):
+        value = super(CUPostalCodeField, self).to_python(value)
+        return value.strip()
+
+
+class CUIdentityCardNumberField(RegexField):
+    """
+    A form field for a cuban identity card number.
+
+    Taken from : http://www.postdata.club/issues/201609/es-usted-unico-en-cuba.html
+
+    The cuban identity card number is generated by a mathematical algorithm following those rules:
+    - Combination of 11 digits.
+    - The first 6 digits represents the birthday of the owner.
+        -> '%y%m%d' format, ex: `860619`
+    - 7th digit represent the century.
+        -> 9 for XIX
+        -> 0-5 for XX
+        -> 6, 7 and 8 for XXI
+    - 8th, 9th and 11th aleatory.
+    - 10th represent the sex of the owner. Male for the even numbers and Female for odd numbers.
+
+    .. versionadded:: 1.6
+    """
+
+    default_error_messages = {
+        'invalid': _('Enter a valid identity card number in the format XXXXXXXXXXX.'),
+    }
+
+    def __init__(self, *args, **kwargs):
+        super(CUIdentityCardNumberField, self).__init__(r'^\d{11}$', *args, **kwargs)
+
+    def _set_regex(self, regex):
```

Comment:
You shouldn't override `_set_regex`. Either use `CharField` and set the `CUIdentityCardNumberValidator` using the `validators` arg in `__init__()` or use the `RegexField` like you do now and check the date in `clean()` after calling `super` instead of using the `CUIdentityCardNumberValidator`.